BSP/STM32F413/Src/BSP_Contactor.c

```diff
@@ -2,41 +2,110 @@
+/**
+ * @brief   Initializes the GPIO pins that interfaces with the Contactors.
  *          Two GPIO pins are initialized. One as an output and one as an input.
  *          The output pin controls the state and the input pin views what state the contactor is through the
  *          Aux pins.
  * @param   None
  * @return  None
  */
 void BSP_Contactor_Init(void) {
-    GPIO_InitTypeDef GPIO_InitStruct;
+    GPIO_InitTypeDef GPIO_C1Init;
+	GPIO_InitTypeDef GPIO_C2Init;
+	GPIO_InitTypeDef GPIO_C3Init;
 	
 	// Initialize clock
-	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOB, ENABLE);
+	RCC_AHB1PeriphClockCmd(C1_PERIPH, ENABLE);
+	RCC_AHB1PeriphClockCmd(C2_PERIPH, ENABLE);
+	RCC_AHB1PeriphClockCmd(C3_PERIPH, ENABLE);
 	
-	// Initialize PB0 as output
-	GPIO_InitStruct.GPIO_Pin = GPIO_Pin_0;			// Pin 0 is output
-	GPIO_InitStruct.GPIO_Mode = GPIO_Mode_OUT;
-	GPIO_InitStruct.GPIO_Speed = GPIO_Speed_50MHz;
-	GPIO_InitStruct.GPIO_OType = GPIO_OType_PP;
-	GPIO_InitStruct.GPIO_PuPd = GPIO_PuPd_UP;
-	GPIO_Init(GPIOB, &GPIO_InitStruct);
-	GPIO_InitStruct.GPIO_Pin = GPIO_Pin_1;			// Pin 1 is input
-	GPIO_InitStruct.GPIO_Mode = GPIO_Mode_IN;
-	GPIO_Init(GPIOB, &GPIO_InitStruct);
+	// Initialize Port B for Contactor 1
+	// PB0 - output
+	// PB1 - input
+	GPIO_C1Init.GPIO_Pin = GPIO_Pin_0;
+	GPIO_C1Init.GPIO_Mode = GPIO_Mode_OUT;
+	GPIO_C1Init.GPIO_Speed = GPIO_Speed_50MHz;
+	GPIO_C1Init.GPIO_OType = GPIO_OType_PP;
+	GPIO_C1Init.GPIO_PuPd = GPIO_PuPd_UP;
+	GPIO_Init(C1_PORT, &GPIO_C1Init);
+	GPIO_C1Init.GPIO_Pin = GPIO_Pin_1;
+	GPIO_C1Init.GPIO_Mode = GPIO_Mode_IN;
+	GPIO_Init(C1_PORT, &GPIO_C1Init);
+
+	// Initialize Port A for Contactor 2
+	// PA4 - output
+	// PA5 - input
+	GPIO_C2Init.GPIO_Pin = GPIO_Pin_4;
+	GPIO_C2Init.GPIO_Mode = GPIO_Mode_OUT;
+	GPIO_C2Init.GPIO_Speed = GPIO_Speed_50MHz;
+	GPIO_C2Init.GPIO_OType = GPIO_OType_PP;
+	GPIO_C2Init.GPIO_PuPd = GPIO_PuPd_UP;
+	GPIO_Init(C2_PORT, &GPIO_C2Init);
+	GPIO_C2Init.GPIO_Pin = GPIO_Pin_5;
+	GPIO_C2Init.GPIO_Mode = GPIO_Mode_IN;
+	GPIO_Init(C2_PORT, &GPIO_C2Init);
+
+	// Initialize Port C for Contactor 3
+	// PC0 - output
+	// PC1 - input
+	GPIO_C3Init.GPIO_Pin = GPIO_Pin_0;
+	GPIO_C3Init.GPIO_Mode = GPIO_Mode_OUT;
+	GPIO_C3Init.GPIO_Speed = GPIO_Speed_50MHz;
+	GPIO_C3Init.GPIO_OType = GPIO_OType_PP;
+	GPIO_C3Init.GPIO_PuPd = GPIO_PuPd_UP;
+	GPIO_Init(C3_PORT, &GPIO_C3Init);
+	GPIO_C3Init.GPIO_Pin = GPIO_Pin_1;
+	GPIO_C3Init.GPIO_Mode = GPIO_Mode_IN;
+	GPIO_Init(C3_PORT, &GPIO_C3Init);
 }
 
 /**
- * @brief   Closes the Contactor switch i.e. turns on the whole electrical system.
+ * @brief   Closes all Contactor switch i.e. turns on the whole electrical system.
  * @note    May be good in the future to make this return something if the contactor could not successfully close.
  * @param   None
  * @return  None
  */
-void BSP_Contactor_On(void) {
-    GPIO_WriteBit(GPIOB, GPIO_Pin_0, Bit_SET);      // Set pin 0 high
+void BSP_Contactor_On(Contactor c) {
+	// set output pins HIGH
+	if (c & ARRAY_CONTACTOR)
```

Comment:
having Contactor c as input implies only one input at a time instead of the multi-input bit mapping scheme here; 
could be more readable if input is an integer instead of an enum (because ORing enums probably not the best practice)